pkgs/development/python-modules/fastmri/default.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pythonOlder,
+
+  # build system
+  setuptools,
+  setuptools-scm,
+
+  # dependencies
+  numpy,
+  scikit-image,
+  torchvision,
+  torch,
+  runstats,
+  pytorch-lightning,
+  h5py,
+  pyyaml,
+  torchmetrics,
+  pandas,
+
+  # tests
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "fastmri";
+  version = "0.3.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "facebookresearch";
+    repo = "fastMRI";
+    tag = "v${version}";
+    hash = "sha256-0IJV8OhY5kPWQwUYPKfmdI67TyYzDAPlwohdc0jWcV4=";
+  };
+
+  # banding_removal folder also has a subfolder named "fastmri"
+  # and np.product is substituted with np.prod in new numpy versions
+  postPatch = ''
+    substituteInPlace tests/test_math.py \
+      --replace "np.product" "np.prod"
+    substituteInPlace tests/conftest.py \
+      --replace "np.product" "np.prod"
+
+    rm -rf banding_removal
+  '';
+
+  build-system = [
+    setuptools
+    setuptools-scm
+  ];
+
+  dependencies = [
+    numpy
+    scikit-image
+    torchvision
+    torch
+    runstats
+    pytorch-lightning
+    h5py
+    pyyaml
+    torchmetrics
+    pandas
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  disabledTestPaths = [
+    # much older version of pytorch-lightning is used
+    "tests/test_modules.py"
+  ];
+
+  pythonImportsCheck = [ "fastmri" ];
+
+  meta = {
+    description = "A large-scale dataset of both raw MRI measurements and clinical MRI images.";
```

Comment:
I think this is not quite right despite the Github repo description, since this package does not actually include the dataset (which is not MIT licensed), but rather related tooling?

Reply:
Ah thank you, added an explanation of my own